### PR TITLE
Demo webpack app + CLI node app

### PR DESCRIPTION
**Pull request by QnnOkabayashi**
This PR builds off of #208 by adding a basic webpack demo that bundles everything up and lets you use the demo ICU4X bindings in `example/js/lib/` on a webpage. To use, navigate to `example/js/app-webpack/`, run `npm install`, and then `npm run build` to run the bundler. Then `index.html` can be viewed through the browser (I use [LiveServer](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer). Additionally, `npm run clean` removes all the auto generated files.

Additionally, I changed `example/js/app` to instead offer a similar web interface to `app-webpack`, and can be set up in the same way, where the only difference is that it doesn't bundle everything together nicely. I also replaced the ava tests in `app/` with a toy CLI tool that lets you format numbers. The ava tests are still in `lib/`.

Finally, this fixes a few small bugs encountered along the way, like how `DiplomatBuf` finalization code was broken and untested (as also found in https://github.com/unicode-org/icu4x/pull/2216).

### CODE REVIEW

**Review by Manishearth — COMMENTED** [review]
looks good, defer to @sffc's review

Could you also write documentation for how to use all of this (and what needs to be done manually?). Can be separate.

**Review by Manishearth — APPROVED** [review]
I kinda meant documentation on how to build such a package for your own diplomat usage, but that could also go in the diplomat book or elsewhere in the crate, this is useful too

**Comment by QnnOkabayashi** [comment]
> I kinda meant documentation on how to build such a package for your own diplomat usage, but that could also go in the diplomat book or elsewhere in the crate, this is useful too

That's gonna require some more work, probably involving cargo make.

**Comment by Manishearth** [comment]
@QnnOkabayashi how would it involve cargo-make? I envision cargo-make being useful for streamlining CI and development tasks, but if I as a library maintainer (of say, ICU4X), want to make an ICU4X npm package, I'm going to have to cobble together diplomat-tool calls and some copied-over files somehow, and it would be nice to have docs about that

**Comment by QnnOkabayashi** [comment]
Oh I see. I can definitely ad that / make that experience better, but probably should wait for the next PR to not bloat this one.

**Comment by Manishearth** [comment]
Anyway, we can land this when @sffc approves

**Review by Manishearth — APPROVED** [review]

File: example/js/lib/api/diplomat-runtime.js
Comment (Manishearth) [inline example/js/lib/api/diplomat-runtime.js:83]: nit: i think we shoudl have a function in DiplomatRuntime that can handle this

File: example/js/app-webpack/package.json
Comment (sffc) [inline example/js/app-webpack/package.json:8]: Suggestion: try `npm run webpack` or `./node_modules/bin/webpack`. `npx` is good for one-off commands on the command line, but I think it's not what you want in a build script, because you should instead download your dependencies in devDependencies and then have this all be local.

File: example/js/lib/api/diplomat-runtime.js
Comment (sffc) [inline example/js/lib/api/diplomat-runtime.js:87]: Suggestion: put a similar check into finalization registries that are auto-generated, not just DiplomatBuf

EDIT: Nevermind; this appears to be the only place we use `unregister`. Hopefully having this code here will mean that if we add more `unregister` in the future then this will get copied into place.

**Review by sffc — APPROVED** [review]

File: example/js/lib/api/diplomat-runtime.js
Comment (QnnOkabayashi) [reply example/js/lib/api/diplomat-runtime.js:83]: This is the only place where we ever unregister, I think this should be fine? I also don't anticipate we add anything else where we might want to manually free

Comment (Manishearth) [reply example/js/lib/api/diplomat-runtime.js:83]: Yeah my reasoning is more so that the generated code can be as compact as possible

Comment (Manishearth) [reply example/js/lib/api/diplomat-runtime.js:83]: `this.free = make_buf_unregisterer(this)` or something

**Pull request merged**
